### PR TITLE
perf: patch libuv to use posix_spawn on macOS

### PR DESCRIPTION
**Pull request by pdesantis**
#### Description of Change

Spawning child processes in an Electron application with a hardened runtime has become slow in macOS Big Sur.

This patch is a squashed version of https://github.com/libuv/libuv/pull/3064, and should be removed when the libuv PR is merged.

Fixes: https://github.com/libuv/libuv/issues/3050
Fixes: https://github.com/electron/electron/issues/26143
PR-URL: https://github.com/libuv/libuv/pull/3064

#### Special thanks
@jpcanepa: authoring the libuv patch
@davidzech @marcello3d @srubin: identifying the underlying Electron issue
@marcello3d: creating the [Electron repro](https://github.com/descriptinc/electron-spawn-perf-bug)
@bnoordhuis: assistance with libuv

#### Stakeholders
@deepak1556: thanks for the [libuv repro](https://github.com/deepak1556/libuv-spawn)
@erickzhao: thanks for the support in the Electron Discord & assisting with Electron build-tools

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `PATH` env var breaking change is fixed
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed slow child process spawning on macOS Big Sur

### CODE REVIEW

**Comment by nornagon** [comment]
Am I correct in understanding that the breaking change is that previously, the `PATH` in the `env` option was used to resolve the executable, but now the `PATH` of the parent process is used? That seems potentially dangerous, but I'll leave that to the libuv maintainers to comment on. We should hold on merging this until we clarify whether we feel OK about merging that breaking change or if there's a way to restore the old behavior and make this non-breaking.

**Comment by pdesantis** [comment]
@nornagon thanks for taking a look! Correct, that's the breaking change I was referring to (I'll update the description to clarify). In my opinion this should not be merged until that is fixed. My motivations for the draft PR are:
1. This has fixed the Big Sur issue for us, hopefully some others in the community find it useful.
2. Once the blocker is resolved, is this the correct way to patch Electron?

**Comment by nornagon** [comment]
@pdesantis yes, this is the right way to land a patch in Electron.

**Comment by codebytere** [comment]
@pdesantis do you still plan to work on this?

**Comment by pdesantis** [comment]
@codebytere yes! I'm just waiting on final approval for https://github.com/libuv/libuv/pull/3064 (which we submitted yesterday), then will update this PR.

**Comment by pdesantis** [comment]
@codebytere @nornagon I'm marking this as Ready for Review now. We fixed the aforementioned breaking change and have received 1 merge approval on libuv PR https://github.com/libuv/libuv/pull/3064 (but it has not been merged yet).

I'm guessing you want to wait for libuv to merge the PR, but figured now is a good time to start the Electron review process.

I had to make 1 change to the patch (stripping trailing whitespace) to pass Electron's lint. I can squash this down into 1 commit once the review is done.

**Review by nornagon — APPROVED** [review]
I'm OK merging / backporting this once it's merged upstream.

**Comment by pdesantis** [comment]
> I'm OK merging / backporting this once it's merged upstream.

Awesome, I'll report back once it's merged. In the meantime, what should I do about these test failures?

**Comment by jkleinsc** [comment]
@pdesantis the patch needs to be updated.  If you are using electron build tools, you can run `e patches node` on your local checkout which should update the patch locally.  Alternatively, you can go to one of the failed jobs, eg https://app.circleci.com/pipelines/github/electron/electron/35194/workflows/fe990252-b819-4d60-91ce-833802f7b2ee/jobs/771654, click on the Artifacts tab, and download the file patches/update-patches.patch to your electron dir and then run `git am update-patches.patch`

**Comment by andrejgloot** [comment]
https://github.com/libuv/libuv/pull/3064 PR has been merged and closed now

**Comment by pdesantis** [comment]
Thanks @jkleinsc, done!

I'm unsure about the failed Windows test: `not ok 398 <webview> tag did-change-theme-color event emits when theme color changes`. The libuv changes only affect macOS. Is it possible that's just a flaky test?

File: patches/node/darwin_libuv_use_posix_spawn.patch
Comment (deepak1556) [inline patches/node/darwin_libuv_use_posix_spawn.patch:11]: > with the addition of API availability annotations to fix a build warning (since Electron compiles with the `-Wunguarded-availability-new` flag)

Looking at the patch, this comment is no longer valid. Right ?

**Review by deepak1556 — APPROVED** [review]
Failing test is unrelated.

File: patches/node/darwin_libuv_use_posix_spawn.patch
Comment (pdesantis) [reply patches/node/darwin_libuv_use_posix_spawn.patch:11]: Good catch, I'll update the comment!

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**
